File: golang/primes.go
package main

import "fmt"

func isPrime(x int) bool {
  for k := 2; k < x; k++{
		if x % k == 0 {
			return false
		}
	}
	return true
}

func main() {
  counter := 0
	for k := 2; k < 100000; k++ {
		if isPrime(k) {
        counter++
			}
	}
  fmt.Print("Primes:")
  fmt.Print( counter)
}

File: pascal/primes.pas
program Primes;
var
k : LongInt;

function isPrime(current : LongInt) : boolean;
var
l : LongInt;
modulus : LongInt;
begin
  isPrime := true;
  for l := 2 to current - 1 do
  begin
        modulus := current mod l;
        if modulus = 0 then
          begin
                isPrime := false;break
          end
  end;
end;
var
counter : LongInt;
BEGIN
    counter := 0;
    for k := 2 to 100000 do
    begin
        if isPrime(k) then
                counter := counter + 1;
    end;
    writeln('primes');
    writeln(counter)



END.
